File: zendrive-api/src/main/java/com/zendrive/api/rest/model/dto/scan/local/LocalScanDto.java
package com.zendrive.api.rest.model.dto.scan.local;

import com.zendrive.api.core.model.metafile.MetaFileConfig;
import com.zendrive.api.core.model.metafile.Permissions;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class LocalScanDto {
    private String destinationId;
    private Permissions permissions;
    private MetaFileConfig config;
}

File: zendrive-ui/src/api/models/scan/local/LocalScanDto.ts
import { MetafileConfig } from "@apiModels/metafile";
import { MetafilePermissions } from "@apiModels/metafile/MetafilePermissions";

export class LocalScanDto {
	destinationId?: string;
	permissions: MetafilePermissions;
	config: MetafileConfig;

	constructor(destinationId: string, permissions: MetafilePermissions, config: MetafileConfig) {
		this.destinationId = destinationId;
		this.permissions = permissions;
		this.config = config;
	}
}
